internal/controllers/machinedeployment/machinedeployment_status_test.go

```diff
@@ -0,0 +1,1185 @@
+			machineSets: []*clusterv1.MachineSet{
+				fakeMachineSet("ms1"),
+			},
+			expectReadyReplicas:     nil,
+			expectAvailableReplicas: nil,
+			expectUpToDateReplicas:  nil,
+		},
+		{
+			name: "MachineSets with replicas set",
+			machineSets: []*clusterv1.MachineSet{
+				fakeMachineSet("ms1", withStatusV1beta2ReadyReplicas(5), withStatusV1beta2AvailableReplicas(3), withStatusV1beta2UpToDateReplicas(3)),
+				fakeMachineSet("ms2", withStatusV1beta2ReadyReplicas(2), withStatusV1beta2AvailableReplicas(2), withStatusV1beta2UpToDateReplicas(1)),
+				fakeMachineSet("ms3"),
+			},
+			expectReadyReplicas:     ptr.To(int32(7)),
+			expectAvailableReplicas: ptr.To(int32(5)),
+			expectUpToDateReplicas:  ptr.To(int32(4)),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			md := &clusterv1.MachineDeployment{}
+			setReplicas(md, tt.machineSets)
+
+			g.Expect(md.Status.V1Beta2).ToNot(BeNil())
+			g.Expect(md.Status.V1Beta2.ReadyReplicas).To(Equal(tt.expectReadyReplicas))
+			g.Expect(md.Status.V1Beta2.AvailableReplicas).To(Equal(tt.expectAvailableReplicas))
+			g.Expect(md.Status.V1Beta2.UpToDateReplicas).To(Equal(tt.expectUpToDateReplicas))
+		})
+	}
+}
+
+func Test_setAvailableCondition(t *testing.T) {
+	tests := []struct {
+		name                                         string
+		machineDeployment                            *clusterv1.MachineDeployment
+		getAndAdoptMachineSetsForDeploymentSucceeded bool
+		expectCondition                              metav1.Condition
+	}{
+		{
+			name:              "getAndAdoptMachineSetsForDeploymentSucceeded failed",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			getAndAdoptMachineSetsForDeploymentSucceeded: false,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineDeploymentAvailableV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.MachineDeploymentAvailableInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:              "spec.replicas not set",
+			machineDeployment: &clusterv1.MachineDeployment{},
+			getAndAdoptMachineSetsForDeploymentSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineDeploymentAvailableV1Beta2Condition,
+				Status: metav1.ConditionUnknown,
+				Reason: clusterv1.MachineDeploymentAvailableWaitingForReplicasSetV1Beta2Reason,
+			},
+		},
+		{
+			name: "status.v1beta2 not set",
+			machineDeployment: &clusterv1.MachineDeployment{
+				Spec: clusterv1.MachineDeploymentSpec{Replicas: ptr.To(int32(5))},
+			},
+			getAndAdoptMachineSetsForDeploymentSucceeded: true,
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.MachineDeploymentAvailableV1Beta2Condition,
+				Status: metav1.ConditionUnknown,
+				Reason: clusterv1.MachineDeploymentAvailableWaitingForAvailableReplicasSetV1Beta2Reason,
+			},
+		},
+		{
+			name: "all the expected replicase are available",
```

Comment:
```suggestion
			name: "all the expected replicas are available",
```
(there are a few more occurences below)